agent/agent_worker.go

```diff
@@ -468,22 +470,16 @@ func (a *AgentWorker) runPingLoop(ctx context.Context, idleMon *idleMonitor) err
 		}
 
 		ranJob = true
-		if a.agentConfiguration.DisconnectAfterJob {
-			// Unless paused, this agent disconnects after the next ping.
-			// Do the ping immediately so we reduce the chances our agent is assigned a job
-			pingTicker.Reset(pingInterval)
-			continue
-		}
```

Comment:
<img width="1292" height="622" alt="Image" src="https://github.com/user-attachments/assets/d1bed69d-2c5e-432c-8217-49cb8de813f5" />

hmm looks like this is at least load bearing to some tests currently, so we'll need to rework that little bit